src/Internal/Requirements/GoogleProductFeedValidator.php

```diff
@@ -0,0 +1,100 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Internal\Requirements;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Exception\ExtensionRequirementException;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\MerchantStatuses;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\TransientsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\PluginHelper;
+use DateTime;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class GoogleProductFeedValidator
+ *
+ * @since x.x.x
+ *
+ * @package AutomatticWooCommerceGoogleListingsAndAdsInternalRequirements
+ */
+class GoogleProductFeedValidator extends RequirementValidator {
+
+	use PluginHelper;
+
+	/**
+	 * Validate all requirements for the plugin to function properly.
+	 *
+	 * @return bool
+	 */
+	public function validate(): bool {
+		try {
+			$this->validate_google_product_feed_inactive();
+		} catch ( ExtensionRequirementException $e ) {
+
+			add_filter(
+				'woocommerce_gla_account_issues',
+				function( $issues, $current_time ) {
+					return $this->add_conflict_issue( $issues, $current_time );
+				},
+				10,
+				2
+			);
+
+			add_filter(
+				'deactivated_plugin',
+				function( $plugin ) {
+					if ( 'woocommerce-product-feeds/woocommerce-gpf.php' === $plugin ) {
+						woogle_get_container()->get( MerchantStatuses::class )->clear_cache();
```

Comment:
This feels odd, but I suppose it's the way to do it since as you mentioned, the container dependencies aren't loaded yet when we do these validation checks.

Do we need any additional validation that our plugin is indeed active and the classes are usable? I know we run this validation check as last, but in theory if we would run it in a different order (or add another validation class later on), we would hook into `deactivated_plugin` and return true. So any followup validations could prevent the services from loading.

Reply:
The other option was to simply use `delete_transient` with a hard-coded name (using `PluginHelper::get_slug()`), but this seemed  a bit more elegant.

I've added a quick check to make sure that that the container `get()` call returns the desired class before clearing the cache. If subsequent validations block the services from being registered, this should prevent a `clear_cache` call using a null variable. b4be28a113807811d069a0af19ee76f052c21921